### DISCUSSION TITLE
Avoid conflicts between system fields and custom fields

### DISCUSSION THREAD

**Discussion by joselcvarela** [directus/directus#8991] [Feature Requests]
Hello all,

https://github.com/directus/directus/blob/8b362c038326ef2979d05ac628836d507ed07b39/api/src/database/migrations/20210903A-add-auth-provider.ts#L8-L9

Here we are adding a new field into `directus_users` system collection. But, what if a project already has this field?
We allow people to add new fields to system tables. Although, when we also want to add new fields, some existing projects wouldn't be able to migrate, because name conflict.

One solution could be adding a prefix or suffix to system fields, and then prevent people to create fields with that prefix/suffix.

**Comment by benhaynes** [comment]
If you are adding a custom field to a system table, and the field has the same name... couldn't you just use the system one? Do you have an example of a custom field that would be added? I'm thinking of `directus_files.tags`, but maybe you have a more specific use-case.

**Reply by joselcvarela** [reply]
This appeared while I was reviewing a PR. Let's take a look into it to provide more context.
https://github.com/directus/directus/pull/8660/files#diff-84b19bc7b40f836ba7d897eede62e65e97e74ab5cd5a607b098615ab9edd11e4

Here you can see that a new field `auth_data` is being created on `directus_users`.
But if an existent project already has this field on this table, it wouldn't be able to upgrade since migration will fail because of that field already exists.

**Reply by benhaynes** [reply]
Ahh, gotcha. I'd rather not add a prefix to all system fields, but we can discuss. I don't think we'll have too many _new_ fields added to the editable system tables (users/files), but this is an important consideration, as it would "technically" make us _adding_ any new core system fields a breaking change.

**Reply by rijkvanzanten** [reply]
I think we'll have to file this under user risk. We can add a note in the docs that it's recommended to add a prefix to the custom fields, as we might change the core fields int he future 👍🏻

**Comment by micheljacquot34** [comment]
I had the problem when migrating from rc96 to above versions. 
We use a table called collections
and a M2O in another table called collection. 

And also a similar issue, using a collection named latest

```js
{
  "errors": [
    {
      "message": "update \"directus_fields\" set \"group\" = $1 where \"group\" = $2 and \"collection\" = $3 - invalid input syntax for type integer: \"latest_id\"",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
